packages/block-editor/src/components/block-preview/index.js

```diff
@@ -33,19 +37,57 @@ function BlockPreview( props ) {
 }
 
 export function BlockPreviewContent( { blocks, settings } ) {
+	const previewRef = useRef( null );
+
+	const [ previewScale, setPreviewScale ] = useState( 1 );
+	const [ visibility, setVisibility ] = useState( 'hidden' );
+
+	let timerId;
+
+	// Dynamically calculate the scale factor
+	useLayoutEffect( () => {
+		const { clientId } = blocks;
+		timerId = setTimeout( () => {
```

Comment:
Let's add a comment to explain why we need the setTimeout (the block list being rendered async..)

Reply:
It's pretty grim and I felt bad when I added this in my original PR. Is there no way to detect the `rendered` event?

Reply:
The solution I proposed (see the PR comment) solve some of it but I think the block list is rendered twice. First at init with an empty block list, and then a second time with the full block list. that's because of how `BlockEditorProvider` works. (init on mount). I think that's the main issue here. So when first rendered, the block list is empty.

Reply:
Is there no way of propagating the change of state within `BlockEditorProvider` from "empty" to "I have some blocks" via a callback or something?